src/auth/strategies/at.strategy.ts

```diff
@@ -34,20 +34,28 @@ export class AtStrategy extends PassportStrategy(Strategy, "jwt-at") {
     }
 
     async validate(payload: any) {
-        const userInDb = await this.usersService.getUserRolesById(payload.sub);
+        //Check if user id provided in payload exists in db
+        const userInDb = await this.usersService.findUserById(payload.sub);
```

Comment:
The problem with this approach is this is another extra db query on a semi frequent operation.

Is it possible to use the original query but throw an error if `!userInDb`? I think it returns null when nothing is found

https://www.prisma.io/docs/orm/reference/prisma-client-reference#findunique

Reply:
The original query uses only getUserRolesById which will return only the formatted user roles even when it finds no user in the db: it will just return {roles:undefined} for example. We could refactor getUserRolesById to only return the roles when the user actually exists, that would indeed avoid the extra query

Reply:
yes I think that's a better option, we won't need to return `roles:undefined` if there's no user. And yeah that makes sense I guess because we have a select, so it would return those instead of a null on a basic findUnique